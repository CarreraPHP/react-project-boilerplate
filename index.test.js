const { execSync } = require('child_process')
const rimraf = require('rimraf')
const { log, error } = require('./helpers/logger')

// TODO: Jest can't process coverage of spaned processes
// May need to wrap NYC to get the coverage of all the
// code executed here

describe('Integration', () => {
  beforeEach(() => {
    // Cleanup the temp
    rimraf.sync('./tmp', {}, () => {
      log('No tmp directory to remove.')
    })
  })

  it('Can generate a new project from the default template', async () => {
    // Run the generator expecting successful STDOUT
    try {
      await execSync('node ./index.js --no-git -- tmp', { stdio: 'inherit' })
    } catch (e) {
      error('Failed to complete generation process.')
      expect(true).toEqual(false) // Force test failure
    }
    // Cleanup the temp which verifies the tmp folder got populated
    rimraf.sync('./tmp', {}, () => {
      error('No tmp directory to remove.')
      expect(true).toEqual(false) // Force test failure
    })
  })
})
