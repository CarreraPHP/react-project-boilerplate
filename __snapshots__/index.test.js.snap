// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration Test Generated App New component wizard(plop) can generate new atoms through the wizard 1`] = `
"export { default } from './MockComponent'
"
`;

exports[`Integration Test Generated App New component wizard(plop) can generate new atoms through the wizard 2`] = `
"import React from 'react'

const MockComponent = () => {
  return(
    <>
      <p>Boilerplate mock component component</p>
    </>
  )
}

export default MockComponent
"
`;

exports[`Integration Test Generated App New component wizard(plop) can generate new atoms through the wizard 3`] = `
"import React from 'react'
import TestRenderer from 'react-test-renderer'
import MockComponent from '.'

describe('MockComponent', () => {
  it('creates short description', () => {
    const result = TestRenderer.create(
      <MockComponent />
    ).toJSON()

    expect(result).toMatchSnapshot()
  })
})
"
`;

exports[`Integration Test Generated App New component wizard(plop) can generate new molecules through the wizard 1`] = `
"export { default } from './MockComponent'
"
`;

exports[`Integration Test Generated App New component wizard(plop) can generate new molecules through the wizard 2`] = `
"import React from 'react'

const MockComponent = () => {
  return(
    <>
      <p>Boilerplate mock component component</p>
    </>
  )
}

export default MockComponent
"
`;

exports[`Integration Test Generated App New component wizard(plop) can generate new molecules through the wizard 3`] = `
"import React from 'react'
import TestRenderer from 'react-test-renderer'
import MockComponent from '.'

describe('MockComponent', () => {
  it('creates short description', () => {
    const result = TestRenderer.create(
      <MockComponent />
    ).toJSON()

    expect(result).toMatchSnapshot()
  })
})
"
`;

exports[`Integration Test Generated App New component wizard(plop) can generate new organisms through the wizard 1`] = `
"export { default } from './MockComponent'
"
`;

exports[`Integration Test Generated App New component wizard(plop) can generate new organisms through the wizard 2`] = `
"import React from 'react'

const MockComponent = () => {
  return(
    <>
      <p>Boilerplate mock component component</p>
    </>
  )
}

export default MockComponent
"
`;

exports[`Integration Test Generated App New component wizard(plop) can generate new organisms through the wizard 3`] = `
"import React from 'react'
import TestRenderer from 'react-test-renderer'
import MockComponent from '.'

describe('MockComponent', () => {
  it('creates short description', () => {
    const result = TestRenderer.create(
      <MockComponent />
    ).toJSON()

    expect(result).toMatchSnapshot()
  })
})
"
`;
